### PR TITLE
Added module arriba

### PR DESCRIPTION
**Pull request by praveenraj2018**
Added a new module arriba for fusion detection.

### CODE REVIEW

File: modules/arriba/main.nf [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+// TOOL DESCRIPTION:
+// Arriba is a command-line tool for the detection of gene fusions from RNA-Seq data. It was developed for the use in a clinical research setting.
+// Apart from gene fusions, Arriba can detect other structural rearrangements with potential clinical relevance, such as viral integration sites,
+// internal tandem duplications, whole exon duplications, truncations of genes (i.e., breakpoints in introns and intergenic regions).
+// Source: https://github.com/suhrig/arriba
+// Article: Sebastian Uhrig, Julia Ellermann, Tatjana Walther, Pauline Burkhardt, Martina Fröhlich, Barbara Hutter, Umut H. Toprak, Olaf Neumann, Albrecht Stenzinger, Claudia Scholl, Stefan Fröhling and Benedikt Brors: Accurate and efficient detection of gene fusions from RNA sequencing data. Genome Research.
+// DOI: 10.1101/gr.257246.119
```
Comment (drpatelh) [inline modules/arriba/main.nf:10]: This should all be moved to `meta.yml` where appropriate.

File: modules/arriba/main.nf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    path fasta
+    path gtf
+
+    output:
+    tuple val(meta), path("*_fusions.tsv")       , emit: tsv
+    path "*.version.txt"                         , emit: version
```
Comment (drpatelh) [inline modules/arriba/main.nf:36]:
```suggestion
    tuple val(meta), path("*.tsv"), emit: tsv
    path "*.version.txt"          , emit: version
```

File: modules/arriba/main.nf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    script:
+    def software            = getSoftwareName(task.process)
+    def prefix              = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def blacklist_option    = (options.args.contains('-b')) ? '' : '-f blacklist'
+
```
Comment (drpatelh) [inline modules/arriba/main.nf:42]:
```suggestion
    def software  = getSoftwareName(task.process)
    def prefix    = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
    def blacklist = (options.args.contains('-b')) ? '' : '-f blacklist'
```

File: modules/arriba/main.nf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    arriba \\
+        -x $bam \\
+        -a $fasta \\
+        -g $gtf \\
+        -o ${prefix}_fusions.tsv \\
```
Comment (drpatelh) [inline modules/arriba/main.nf:48]:
```suggestion
        -o ${prefix}.fusions.tsv \\
```

File: modules/arriba/main.nf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        $blacklist_option \\
```
Comment (drpatelh) [inline modules/arriba/main.nf:49]:
```suggestion
        -o ${prefix}.fusions.tsv \\
        $blacklist \\
```

File: modules/arriba/meta.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  - tsv:
+      type: file
+      description: Fusions detected by the algorithm
+      pattern: "*_{fusions.tsv}"
```
Comment (drpatelh) [inline modules/arriba/meta.yml:47]:
```suggestion
      pattern: "*.{fusions.tsv}"
```

File: tests/modules/arriba/main.nf [hunk 7]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { STAR_GENOMEGENERATE } from '../../../modules/star/genomegenerate/main.nf' addParams( options: [args: '--genomeSAindexNbases 11'] )
+include { STAR_ALIGN          } from '../../../modules/star/align/main.nf'          addParams( options: [args: '--readFilesCommand zcat --outSAMtype BAM Unsorted --outSAMunmapped Within --outBAMcompression 0 --outFilterMultimapNmax 50 --peOverlapNbasesMin 10 --alignSplicedMateMapLminOverLmate 0.5 --alignSJstitchMismatchNmax 5 -1 5 5 --chimSegmentMin 10 --chimOutType WithinBAM HardClip --chimJunctionOverhangMin 10 --chimScoreDropMax 30 --chimScoreJunctionNonGTAG 0 --chimScoreSeparation 1 --chimSegmentReadGapMax 3 --chimMultimapNmax 50'] )
+include { ARRIBA }              from '../../../modules/arriba/main.nf'              addParams( options: [:] )
```
Comment (drpatelh) [inline tests/modules/arriba/main.nf:7]:
```suggestion
include { STAR_ALIGN          } from '../../../modules/star/align/main.nf'          addParams( options: [args: '--readFilesCommand zcat --outSAMtype BAM Unsorted --outSAMunmapped Within --outBAMcompression 0 --outFilterMultimapNmax 50 --peOverlapNbasesMin 10 --alignSplicedMateMapLminOverLmate 0.5 --alignSJstitchMismatchNmax 5 -1 5 5 --chimSegmentMin 10 --chimOutType WithinBAM HardClip --chimJunctionOverhangMin 10 --chimScoreDropMax 30 --chimScoreJunctionNonGTAG 0 --chimScoreSeparation 1 --chimSegmentReadGapMax 3 --chimMultimapNmax 50'] )
include { ARRIBA              }              from '../../../modules/arriba/main.nf'              addParams( options: [:] )
```

File: tests/modules/arriba/main.nf [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+workflow test_arriba_se {
```
Comment (drpatelh) [inline tests/modules/arriba/main.nf:9]:
```suggestion
workflow test_arriba_single_end {
```

File: tests/modules/arriba/main.nf [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    fasta = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    gtf = file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true)
+
+    STAR_GENOMEGENERATE ( fasta, gtf )
+    STAR_ALIGN ( input, STAR_GENOMEGENERATE.out.index, gtf )
+    ARRIBA ( STAR_ALIGN.out.bam, fasta, gtf)
+}
+
+workflow test_arriba_pe {
```
Comment (drpatelh) [inline tests/modules/arriba/main.nf:23]:
```suggestion
workflow test_arriba_paired_end {
```

File: modules/arriba/main.nf (hunk 5 above)
Comment (praveenraj2018) [reply modules/arriba/main.nf:49]: I thought of adding -O also to output the discarded fusions (by filters) 
then, the prefix will change as below:

-o ${prefix}.fusions.tsv \\
-O ${prefix}.fusions.discarded.tsv \\

Is this okay?

Comment (drpatelh) [reply modules/arriba/main.nf:49]: Yep, thats perfect 👍🏽

Comment (praveenraj2018) [reply modules/arriba/main.nf:49]: ok. Then the output channels will also change to:
```
output:
    tuple val(meta), path("*.fusions.tsv")              , emit: fusions
    tuple val(meta), path("*.fusions.discarded.tsv")    , emit: fusions_fail
```
Is this the right way to put it?

File: modules/arriba/main.nf (hunk 1 above)
Comment (praveenraj2018) [reply modules/arriba/main.nf]: Done!

File: modules/arriba/main.nf (hunk 2 above)
Comment (praveenraj2018) [reply modules/arriba/main.nf]: Edited.

File: modules/arriba/main.nf (hunk 3 above)
Comment (praveenraj2018) [reply modules/arriba/main.nf]: Done!

File: modules/arriba/meta.yml (hunk 6 above)
Comment (praveenraj2018) [reply modules/arriba/meta.yml]: Done!

File: tests/modules/arriba/main.nf (hunk 7 above)
Comment (praveenraj2018) [reply tests/modules/arriba/main.nf]: Done!

File: tests/modules/arriba/main.nf (hunk 8 above)
Comment (praveenraj2018) [reply tests/modules/arriba/main.nf]: Changed!

File: tests/modules/arriba/main.nf (hunk 9 above)
Comment (praveenraj2018) [reply tests/modules/arriba/main.nf]: changed

File: modules/arriba/main.nf (hunk 5 above)
Comment (drpatelh) [reply modules/arriba/main.nf]: Yep. Exactly.

File: modules/arriba/main.nf [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+    input:
+    tuple val(meta), path(bam)
+    path fasta
+    path gtf
+
+    output:
+    tuple val(meta), path("*.fusions.tsv")              , emit: fusions
+    tuple val(meta), path("*.fusions.discarded.tsv")    , emit: fusions_fail
+    path "*.version.txt"                                , emit: version
```
Comment (drpatelh) [inline modules/arriba/main.nf:29]:
```suggestion
    tuple val(meta), path("*.fusions.tsv")          , emit: fusions
    tuple val(meta), path("*.fusions.discarded.tsv"), emit: fusions_fail
    path "*.version.txt"                            , emit: version
```

File: modules/arriba/main.nf [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+    script:
+    def software  = getSoftwareName(task.process)
+    def prefix    = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def blacklist = (options.args.contains('-b')) ? '' : '-f blacklist'
+
```
Comment (drpatelh) [inline modules/arriba/main.nf:35]:
```suggestion
```

File: tests/modules/arriba/main.nf [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+                [   file(params.test_data['homo_sapiens']['illumina']['test_rnaseq_1_fastq_gz'], checkIfExists: true),
+                    file(params.test_data['homo_sapiens']['illumina']['test_rnaseq_2_fastq_gz'], checkIfExists: true) ]
+            ]
+    fasta = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    gtf = file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true)
+
+    STAR_GENOMEGENERATE ( fasta, gtf )
+    STAR_ALIGN ( input, STAR_GENOMEGENERATE.out.index, gtf )
+    ARRIBA ( STAR_ALIGN.out.bam, fasta, gtf)
```
Comment (drpatelh) [inline tests/modules/arriba/main.nf:35]:
```suggestion
    ARRIBA ( STAR_ALIGN.out.bam, fasta, gtf )
```

File: tests/modules/arriba/main.nf [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+    input = [ [ id:'test', single_end:true ], // meta map
+                [   file(params.test_data['homo_sapiens']['illumina']['test_rnaseq_1_fastq_gz'], checkIfExists: true) ]
+            ]
+    fasta = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    gtf = file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true)
+
+    STAR_GENOMEGENERATE ( fasta, gtf )
+    STAR_ALIGN ( input, STAR_GENOMEGENERATE.out.index, gtf )
+    ARRIBA ( STAR_ALIGN.out.bam, fasta, gtf)
```
Comment (drpatelh) [inline tests/modules/arriba/main.nf:20]:
```suggestion
    ARRIBA ( STAR_ALIGN.out.bam, fasta, gtf )
```

**Review by drpatelh — APPROVED** [review]
Nice! Thanks @praveenraj2018 !

File: modules/arriba/main.nf (hunk 5 above)
Comment (praveenraj2018) [reply modules/arriba/main.nf]: Thanks

**Pull request merged**